hadoop-plugin/src/main/groovy/com/linkedin/gradle/azkaban/yaml/YamlWorkflow.groovy

```diff
@@ -0,0 +1,178 @@
+import com.linkedin.gradle.hadoopdsl.Workflow;
+import com.linkedin.gradle.hadoopdsl.job.Job;
+import com.linkedin.gradle.hadoopdsl.Properties;
+import com.linkedin.gradle.hadoopdsl.job.LaunchJob;
+import com.linkedin.gradle.hadoopdsl.job.StartJob;
+
+/**
+ * Representation of a workflow in .flow file for Azkaban Flow 2.0
+ *
+ * TODO reallocf sort list of nodes into linearization of flow DAG
+ *
+ * Example .flow file:
+ *
+ * ---
+ * config:
+ *   user.to.proxy: azktest
+ *   param.hadoopOutData: /tmp/wordcounthadoopout
+ *   param.inData: /tmp/wordcountpigin
+ *   param.outData: /tmp/wordcountpigout
+ *
+ * nodes:
+ * - name: pigWordCount1
+ *   type: pig
+ *   config:
+ *     pig.script: src/main/pig/wordCountText.pig
+ * - name: hadoopWC1
+ *   type: hadoopJava
+ *   dependsOn:
+ *   - pigWordCount1
+ *   config:
+ *     classpath: ./*
+ *     force.output.overwrite: true
+ *     input.path: ${param.inData}
+ *     job.class: com.linkedin.wordcount.WordCount
+ *     main.args: ${param.inData} ${param.hadoopOutData}
+ *     output.path: ${param.hadoopOutData}
+ * - name: hive1
+ *   type: hive
+ *   config:
+ *     hive.script: src/main/hive/showdb.q
+ * - name: NoOpTest1
+ *   type: noop
+ * - name: hive2
+ *   type: hive
+ *   dependsOn:
+ *   - hive1
+ *   config:
+ *     hive.script: src/main/hive/showTables.sql
+ * - name: java1
+ *   type: javaprocess
+ *   config:
+ *     Xms: 96M
+ *     java.class: com.linkedin.foo.HelloJavaProcessJob
+ * - name: jobCommand1
+ *   type: command
+ *   config:
+ *     command: echo "hello world from job_command_1"
+ * - name: jobCommand2
+ *   type: command
+ *   dependsOn:
+ *   - jobCommand1
+ *   config:
+ *     command: echo "hello world from job_command_2"
+ * ---
+ */
+class YamlWorkflow {
+  String name;
+  String type;
+  List dependsOn;
+  Map config;
+  List nodes;
+
+  /**
+   * Construct YamlJob from Job and the Job's parent scope
+   *
+   * @param workflow The workflow to be converted into Yaml
+   * @param parentScope The parent scope of the workflow
+   * @param subflow Boolean regarding whether or not the workflow is a subflow
+   */
+  YamlWorkflow(Workflow workflow, NamedScope parentScope, boolean subflow) {
+    // Include name/type in yaml only if the workflow is a subflow
+    name = subflow ? workflow.name : null;
+    type = subflow ? "flow" : null;
+    dependsOn = workflow.parentDependencies.toList();
+    config = buildConfig(workflow, parentScope, subflow);
+    nodes = buildNodes(workflow, parentScope);
+  }
+
+  /**
+   * Create the workflow config from the properties in the namespace
+   *
+   * @param workflow The workflow being constructed
+   * @param parentScope The parent scope of the workflow being constructed
+   * @param subflow Boolean regarding whether or not the workflow is a subflow
+   * @return result The map of all properties associated with the workflow
+   */
+  private static Map buildConfig(Workflow workflow, NamedScope parentScope, boolean subflow) {
```

Comment:
How do you handle the scenario where the same property is defined in both the current workflow and the global properties? Should the current flow properties override the global properties?

Reply:
I think in the current implementation the global property overrides the workflow property. This doesn't sound right to me, so I'll change. I'll also include a test to confirm this suspicion. Thanks for thinking of it?

Do you agree that workflow property should override global property? We can do either.

Reply:
Yes, I agree with you that workflow property should override global property.